runtime/src/bank.rs

```diff
@@ -1888,6 +1888,121 @@ pub fn goto_end_of_slot(bank: &mut Bank) {
     }
 }
 
+// The `bank` struct as defined in Solana v0.22.3
+#[derive(Default, Deserialize, Serialize)]
+pub struct LegacyBank0223 {
+    #[serde(skip)]
+    pub rc: BankRc,
+
+    #[serde(skip)]
+    pub src: StatusCacheRc,
+
+    pub blockhash_queue: RwLock<BlockhashQueue>,
+    pub ancestors: HashMap<Slot, usize>,
+    pub hash: RwLock<Hash>,
+    pub parent_hash: Hash,
+    pub parent_slot: Slot,
+
+    #[serde(serialize_with = "serialize_atomicu64")]
+    #[serde(deserialize_with = "deserialize_atomicu64")]
+    pub transaction_count: AtomicU64,
+
+    #[serde(serialize_with = "serialize_atomicu64")]
+    #[serde(deserialize_with = "deserialize_atomicu64")]
+    pub tick_height: AtomicU64,
+
+    #[serde(serialize_with = "serialize_atomicu64")]
+    #[serde(deserialize_with = "deserialize_atomicu64")]
+    pub signature_count: AtomicU64,
+
+    #[serde(serialize_with = "serialize_atomicu64")]
+    #[serde(deserialize_with = "deserialize_atomicu64")]
+    pub capitalization: AtomicU64,
+
+    pub max_tick_height: u64,
+    pub hashes_per_tick: Option<u64>,
+    pub ticks_per_slot: u64,
+    pub ns_per_slot: u128,
+    pub genesis_creation_time: UnixTimestamp,
+    pub slots_per_year: f64,
+    pub slots_per_segment: u64,
+    pub slot: Slot,
+    pub epoch: Epoch,
+    pub block_height: u64,
+    pub collector_id: Pubkey,
+
+    #[serde(serialize_with = "serialize_atomicu64")]
+    #[serde(deserialize_with = "deserialize_atomicu64")]
+    pub collector_fees: AtomicU64,
+
+    pub fee_calculator: FeeCalculator,
+
+    #[serde(serialize_with = "serialize_atomicu64")]
+    #[serde(deserialize_with = "deserialize_atomicu64")]
+    pub collected_rent: AtomicU64,
+
+    pub rent_collector: RentCollector,
+    pub epoch_schedule: EpochSchedule,
+    pub inflation: Arc<RwLock<Inflation>>,
+    pub stakes: RwLock<Stakes>,
+    pub storage_accounts: RwLock<StorageAccounts>,
+    pub epoch_stakes: HashMap<Epoch, Stakes>,
+
+    #[serde(serialize_with = "serialize_atomicbool")]
+    #[serde(deserialize_with = "deserialize_atomicbool")]
+    pub is_delta: AtomicBool,
+
+    pub message_processor: MessageProcessor,
+
+    #[serde(skip)]
+    pub entered_epoch_callback: Arc<RwLock<Option<EnteredEpochCallback>>>,
+
+    #[serde(skip)]
+    pub last_vote_sync: AtomicU64,
+}
+
+impl From<LegacyBank0223> for Bank {
+    fn from(bank: LegacyBank0223) -> Self {
+        Bank {
```

Comment:
Thanks to the rust's ownership, it looks like perfectly forwaring `struct` fields is a piece of case even if myriad of various types are involved. :)

Reply:
yeah, so nice